Assets/Scripts/Layers/LayerTypes/GeoJSONLineLayer.cs

```diff
@@ -39,16 +43,57 @@ protected override void OnLayerActiveInHierarchyChanged(bool activeInHierarchy)
             LineRenderer3D.gameObject.SetActive(activeInHierarchy);
         }
 
-        public void AddAndVisualizeFeature(Feature feature, MultiLineString featureGeometry, CoordinateSystem originalCoordinateSystem)
+        public void AddAndVisualizeFeature<T>(Feature feature, CoordinateSystem originalCoordinateSystem)
+            where T : GeoJSONObject
         {
-            LineFeatures.Add(feature);
-            GeoJSONGeometryVisualizerUtility.VisualizeMultiLineString(featureGeometry, originalCoordinateSystem, LineRenderer3D);
+            // Skip if feature already exists (comparison is done using hashcode based on geometry)
+            if (SpawnedVisualisations.Any(f => f.feature.GetHashCode() == feature.GetHashCode()))
+                return;
+
+            var newFeatureVisualisation = new FeatureLineVisualisations() { feature = feature };
+
+            if (feature.Geometry is MultiLineString multiLineString)
+            {
+                var newLines = GeoJSONGeometryVisualizerUtility.VisualizeMultiLineString(multiLineString, originalCoordinateSystem, lineRenderer3D);
+                newFeatureVisualisation.lines.AddRange(newLines);
+            }
+            else if(feature.Geometry is LineString lineString)
+            {
+                var newLine = GeoJSONGeometryVisualizerUtility.VisualizeLineString(lineString, originalCoordinateSystem, lineRenderer3D);
+                newFeatureVisualisation.lines.Add(newLine);
+            }
+
+            SpawnedVisualisations.Add(newFeatureVisualisation);
         }
 
-        public void AddAndVisualizeFeature(Feature feature, LineString featureGeometry, CoordinateSystem originalCoordinateSystem)
+        /// <summary>
+        /// Checks the Bounds of the visualisations and checks them against the camera frustum
+        /// to remove visualisations that are out of view
+        /// </summary>
+        public void RemoveFeaturesOutOfView()
```

Comment:
Shouldn't this also trigger the stop of Coroutines that are still running in features / visualisations that are still loading?